docs/decisions/0036-text-classification.md

```diff
@@ -0,0 +1,270 @@
+
+```json
+[
+  [
+    {
+      "label": "disappointment",
+      "score": 0.5044490694999695
+    },
+    {
+      "label": "sadness",
+      "score": 0.3469429612159729
+    }
+  ]
+]
+```
+
+## Decision Drivers
+
+If we decide to create a common abstraction for classification models, we should consider the following drivers:
+1. Abstraction should be able to generate classification from different types of data, at least from text and images.
+2. Abstraction should return metadata with generated classification results.
+3. Abstraction should allow parameterize the classification query.
+
+## Considered Options
+
+### Option 1 [Proposed] - With abstraction
+
+This option assumes the creation of a common abstraction layer for classification models.
+
+Since the models return different data structures, we return `IReadOnlyDictionary<string, object?> Result` as a result.
+This type is very abstract by which the consumer must know the structure of the result
+and will probably have to use a cast to a connector-supplied type such like `OpenAIClassificationData`.
+We could just as well return `object` and always require a casting.
+
+```csharp
+public interface ITextClassificationService : IAIService
+{
+    Task<IReadOnlyList<ClassifiedContent>> GetClassifiedContentsAsync(
+        IEnumerable<TextContent> texts,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default);
+}
+
+public interface IImageClassificationService : IAIService
+{
+    Task<IReadOnlyList<ClassifiedContent>> GetClassifiedContentsAsync(
+        IEnumerable<ImageContent> images,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default);
+}
+
+public class ClassifiedContent : KernelContent
+{
+    public ClassifiedContent(
+        IReadOnlyDictionary<string, double> result,
+        object? innerContent = null,
+        string? modelId = null,
+        IReadOnlyDictionary<string, object?>? metadata = null)
+        : base(innerContent, modelId, metadata)
+    {
+        this.Result = result;
+    }
+
+    // value (double) will be used as score value and also as boolean 1|0 (true|false)
+    public IReadOnlyDictionary<string, double> ScoredCategories { get; }
+}
+```
+
+Pros:
+- We have abstraction for classification models.
+
+Cons:
+- double used as decimal value and also as boolean
+
+### Option 2 [Proposed] - With abstraction and common method
+
+Similar to option 1 but with one method with kernelcontent param.
+
+```csharp
+public interface IClassificationService : IAIService
+{
+    Task<IReadOnlyList<ClassifiedContent>> GetClassifiedContentsAsync(
+        IEnumerable<KernelContent> contents,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default);
+
+    IList<Type> SupportedContents { get; }
```

Comment:
This might not be needed for now in the interface as we plan to add in the mid/long term a generics approach to `IService<TIn,TOut>`. 

For the moment we may stick `ITextClassificationService` with the method only to retrieve the contents for `string` as input to start with as this is the current example and we might have `IImageClassificationService` and so forth. 

We had an interesting conversation on the abstractions of Audio abstractions and is was very conclusive the benefits of having `Modal` specific interfaces.

Reply:
Do I understand correctly that the `IService<TIn,TOut>` interfaces will only be wrappers to the current interfaces?